### PR TITLE
fixes issue #22692 Holodeck spawns being sold for infinite cargo cred…

### PR DESCRIPTION
**Pull request by alexkar598**
fix: cargo will not sell holodeck spawned objects

### CODE REVIEW

**Review by Mervill — APPROVED** [review]

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 	var/damtype = "brute"
 	var/force = 0
-
+	var/holospawned = 0
```
Comment (coiax) [inline code/game/objects/objs.dm]: More vars at the /obj level? I don't like that.

Comment (alexkar598) [inline code/game/objects/objs.dm]: Well if i dont make a var at obj level i will have errors since the var will not exist if its not holo

**Comment by coiax** [comment]
Putting a var on the /obj level is SUPER heavyduty for a minor exploit such as this. Holodecks track the objects that they have spawned, have the cargo code check that instead.

**Comment by AnturK** [comment]
Alternative solution would be to check if the objects is on the spawned list of holodeck consoles on export. Might get pricey with the more dense holodeck areas though.

Or if you want to get fancy something that mso proposed : splitting flags into source/meta ones with varedited/holodeck and game ones in the other. This would need few design choices and is a lot more work though.

**Comment by PKPenguin321** [comment]
dont we already have a flag for holodeck items? just check for that flag

**Comment by Cyberboss** [comment]
emagged holoitems are real

**Comment by alexkar598** [comment]
Just saying that coaix said that its a minor issue I won't say it's minor.Cargo can make the difference between malf ai dead or crew dead, head revs dead or head of staff dead , blob dead or crew dead, cult dead or crew dead

**Comment by coiax** [comment]
You're talking about adding a new var to /obj, the root datum of machinery, items, effects, docking ports, structures, singulos. I'm sure @MrStonedOne would have an idea of how many /obj we have existing at any time.

Find another way to fix it.

**Comment by alexkar598** [comment]
Okay since pretty much everyone don't like this fix i will find an other way to fix this issue for now I'm closing this pr

**Comment by MrStonedOne** [comment]
I should point out @coiax that vars only use up memory once changed from their compile time default value.

**Comment by coiax** [comment]
VV is crowded enough as it is. What we need is more flag slots.

**Pull request closed without merging**